libsmtutil/Z3Loader.h

```diff
@@ -0,0 +1,13 @@
+#include <dlfcn.h>
```

Comment:
```suggestion
#pragma once

#include <dlfcn.h>
```

Reply:
And apparently `#define _GNU_SOURCE` should be defined before `dlfcn.h` to enable the gnuism (`dlmopen`).

Reply:
Curious... if I am supposed to do that, I'd have expected the function definition not to be there, if I don't... I'll look into it, the man page sure looks like one should, I just didn't notice that detail.

Reply:
I think gcc/clang by default allow gnu-isms, unless we disable them.

I did that in some project, and it was an awful mess, because half of the libc depended on it.